src/snowcli/cli/sql.py

```diff
@@ -29,40 +58,17 @@ def execute_sql(
         help="File to execute.",
     ),
     **options
-):
+) -> OutputData:
     """
     Executes Snowflake query.
 
     Query to execute can be specified using query option, filename option (all queries from file will be executed)
     or via stdin by piping output from other command. For example `cat my.sql | snow sql`.
     """
-    sys_input = None
-
-    if query and file:
-        raise UsageError("Both query and file provided, please specify only one.")
-
-    if not sys.stdin.isatty():
-        sys_input = sys.stdin.read()
-
-    if sys_input and (query or file):
-        raise UsageError(
-            "Can't use stdin input together with query or filename option."
-        )
-
-    if not query and not file and not sys_input:
-        raise UsageError("Provide either query or filename argument")
-    elif sys_input:
-        sql = sys_input
-    else:
-        sql = query if query else file.read_text()  # type: ignore
-
-    conn = snow_cli_global_context_manager.get_connection()
-
-    results = conn.ctx.execute_string(
-        sql_text=sql,
-        remove_comments=True,
-    )
-    output_data = OutputData()
-    for result in results:
-        output_data.add_data(result)
-    print_output(output_data)
+    cursors = SqlManager().execute(query, file)
```

Comment:
Doesn't it return iterator? Can't we always do
```py
for cursors in cursors:
   output_data.add_cursor(cursor)
```

Reply:
Right